k8s.io/legacy-cloud-providers/gce TestBasePath always fails locally

/sig cloud-provider
/area provider/gcp

run `make test` (or just `make test WHAT=k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce`.

observe failure:

<details>

```
W0604 12:56:56.064945 3036007 gce_disks.go:979] GCE persistent disk "disk" not found in managed zones ()
W0604 12:56:56.080137 3036007 gce_loadbalancer_internal.go:256] clearPreviousInternalResources: could not delete old backend service: fakesvc, err: failed to delete backend service: fakesvc, err: googleapi: got HTTP response code 500 with body: 
W0604 12:56:56.080181 3036007 gce_loadbalancer_internal.go:270] clearPreviousInternalResources(a): more than one health check on the backend service fakesvc, [https://www.googleapis.com/compute/v1/projects/test-project/global/healthChecks/hc1 https://www.googleapis.com/compute/v1/projects/test-project/global/healthChecks/hc2]
W0604 12:56:56.080305 3036007 gce_loadbalancer_internal.go:266] clearPreviousInternalResources: could not delete existing healthcheck: hc1, err: failed to delete health check: hc1, err: googleapi: got HTTP response code 500 with body: 
E0604 12:56:56.104884 3036007 gce_loadbalancer_internal.go:979] forwardingRulesEqual(): failed to parse backend resource URL from existing FR, err - "badBackendService" is not a valid resource URL
I0604 12:56:56.115841 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.115938 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.116018 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.116127 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.116242 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.116299 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.116357 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.116432 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.116454 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.116522 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.116545 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.116561 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.116629 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.116730 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.116755 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.116811 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.118775 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.118925 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.119014 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.119090 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.119185 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.119221 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.119315 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.119350 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.119372 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
E0604 12:56:56.119312 3036007 gce_loadbalancer_external.go:80] ensureExternalLoadBalancer(a(/fakesvc)): Failed to get the desired network tier: unsupported network tier: "SupremeLuxury".
W0604 12:56:56.119484 3036007 gce_loadbalancer_external.go:134] ensureExternalLoadBalancer(a(/fakesvc)): Orphaning static IP  in region us-central1: <nil>.
I0604 12:56:56.119801 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.119896 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.120107 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.120166 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.120201 3036007 gce_loadbalancer_external.go:482] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Deleted target pool.
I0604 12:56:56.120270 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.120357 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.120414 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.120469 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.120487 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.120512 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.120547 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.120572 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.120783 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.120891 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.120971 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.121089 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.121167 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.121229 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.121291 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.121319 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.121380 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.116821 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.121676 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.121399 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.121808 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.122016 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.122077 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.122339 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.122398 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.122435 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.122519 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.122617 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.122736 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.122824 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.122880 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.122518 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.123563 3036007 gce_loadbalancer_external.go:786] LoadBalancer target pool a changed affinity from NONE to ClientIP
I0604 12:56:56.124164 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.124271 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.124363 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.124479 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.124571 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.124693 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.127714 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.127823 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.127914 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.127988 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.128107 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.128174 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.128236 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.128306 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.128330 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.128428 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.128589 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.128680 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.128752 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.128803 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.128892 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.128953 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.129005 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.129027 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.129041 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.129081 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.129114 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.129126 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.129133 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.129206 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
W0604 12:56:56.129297 3036007 gce_loadbalancer_external.go:875] Error parsing firewall SourceRanges: [badSourceRange]
I0604 12:56:56.129323 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.129331 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.129396 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.129393 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.129465 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.129530 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.129554 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.129617 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.129632 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.129648 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.129820 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.129864 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.129968 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.130047 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.130147 3036007 gce_loadbalancer_external.go:753] LoadBalancer port range for forwarding rule a was expected to be 123-123, but was actually 124-124
I0604 12:56:56.130863 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.130952 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.131035 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.131827 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.131912 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.131984 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.132057 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.132147 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.132203 3036007 gce_loadbalancer_external.go:758] LoadBalancer protocol for forwarding rule a was expected to be TCP, but was actually UDP
I0604 12:56:56.132289 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.132396 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.132488 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.132567 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.132674 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.132757 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.132817 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.132846 3036007 gce_loadbalancer_external.go:743] LoadBalancer ip for forwarding rule a was expected to be 1.2.3.0, but was actually 1.2.3.4
I0604 12:56:56.132899 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.132939 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.132995 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.133032 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.133070 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.133091 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.133094 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.133187 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.133272 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.133309 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.133380 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.133382 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.133482 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.133545 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.133603 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.133627 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.133693 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.133713 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.133733 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.133831 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.133901 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.134000 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.134079 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.134143 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
E0604 12:56:56.134183 3036007 gce_healthchecks.go:282] vstring () is not a valid version string: could not parse "" as version
I0604 12:56:56.134257 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.134353 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.134421 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.134478 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.134498 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
E0604 12:56:56.134257 3036007 gce_instances.go:632] Failed to retrieve instances: []
I0604 12:56:56.134594 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.134671 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.134504 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
E0604 12:56:56.135221 3036007 gce_loadbalancer_external.go:465] verifyUserRequestedIP: requested IP "1.1.1.1" for LB test-lb is neither static nor assigned to the LB
I0604 12:56:56.135461 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.139827 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.139960 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.140054 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.141105 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.141191 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.141250 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.141315 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.141343 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.141413 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.141432 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.141451 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.141699 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.141767 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
E0604 12:56:56.136988 3036007 gce_loadbalancer_external.go:449] verifyUserRequestedIP: requested static IP "1.1.1.1" (name: foo) for LB test-lb has network tier Premium, need Standard.
I0604 12:56:56.143450 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.143550 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.143642 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.143729 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.143851 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.143931 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.144005 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.144083 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.144113 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.144193 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.144215 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.144233 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.144447 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.144500 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.135589 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.145054 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.145133 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.135590 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.145200 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.145229 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.145433 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.145491 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
E0604 12:56:56.145630 3036007 gce_instances.go:632] Failed to retrieve instances: [test-node-1]
W0604 12:56:56.149209 3036007 gce_instances.go:568] getFoundInstanceByNames - input instances 3, found 2. Continuing LoadBalancer Update
I0604 12:56:56.145632 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.145647 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.145657 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.145677 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.152701 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.152746 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.152780 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.152789 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.158491 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.158516 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.158547 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.158572 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.158806 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.158871 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.158577 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.159060 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.164997 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.165084 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.165153 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.165234 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.165299 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.165340 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.165390 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.165493 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.178857 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.178933 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.178986 3036007 gce_loadbalancer_external.go:482] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Deleted target pool.
I0604 12:56:56.179056 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.179664 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.179740 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.179813 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.180564 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.180654 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.180673 3036007 gce_loadbalancer_external.go:500] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created initial target pool (now updating the remaining 1 hosts).
I0604 12:56:56.182043 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.182143 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.182223 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.182293 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.182419 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.182487 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.182544 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.182603 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.182622 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.182693 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.182710 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.182728 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
--- FAIL: TestBasePath (0.00s)
    gce_loadbalancer_test.go:190: Compute basePath has changed. Got "https://compute.mtls.googleapis.com/compute/v1/projects/", want "https://compute.googleapis.com/compute/v1/projects/"
I0604 12:56:56.182918 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.182965 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.183100 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.183151 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.183208 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.183283 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.183339 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.183375 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.183435 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.183505 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.184132 3036007 gce_loadbalancer_external.go:504] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Updated target pool (with 1 hosts).
I0604 12:56:56.184725 3036007 gce_loadbalancer_external.go:511] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Updated target pool (with 1 hosts).
I0604 12:56:56.196587 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.196685 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.196760 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.196829 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.196941 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.197003 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.197055 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.197112 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.197131 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.197191 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.197211 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.197227 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.197379 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.200176 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.200743 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.200819 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.200939 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.200994 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.201049 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.201112 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.201170 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.201211 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
I0604 12:56:56.208114 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.208222 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.208305 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.208384 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.208520 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.208603 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.208657 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.208727 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.208749 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.208830 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.208850 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.208869 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.209058 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.209107 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.209441 3036007 gce_loadbalancer_external.go:319] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting firewall rule.
I0604 12:56:56.209465 3036007 gce_loadbalancer_external.go:333] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting IP address.
I0604 12:56:56.209604 3036007 gce_loadbalancer_external.go:337] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting forwarding rule.
I0604 12:56:56.209684 3036007 gce_loadbalancer_external.go:343] ensureExternalLoadBalancerDeleted(a(/fakesvc)): Deleting target pool.
I0604 12:56:56.209737 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check a.
I0604 12:56:56.209818 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-a-http-hc.
I0604 12:56:56.209878 3036007 gce_loadbalancer_external.go:379] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check k8s-test-cluster-id-node.
I0604 12:56:56.209930 3036007 gce_loadbalancer_external.go:401] DeleteExternalTargetPoolAndChecks(a(/fakesvc)): Deleting health check firewall k8s-test-cluster-id-node-http-hc.
E0604 12:56:56.215049 3036007 gce_loadbalancer_internal.go:82] Failed to attach finalizer 'gke.networking.io/l4-ilb-v1' on service /fakesvc - services "fakesvc" not found
I0604 12:56:56.215632 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.215718 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.215797 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.215861 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.215961 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.216022 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.216069 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.216123 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.216142 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.216198 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.216214 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.216229 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.216385 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.216433 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.218920 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.219040 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.219134 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.219215 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.219336 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.219406 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.219480 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.219552 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.219586 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.219656 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.219684 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.219706 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.219901 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.221413 3036007 gce_loadbalancer_external.go:694] Did not find health check test-hc, creating port 12345 path /healthz
I0604 12:56:56.221476 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
I0604 12:56:56.221495 3036007 gce_loadbalancer_external.go:703] Created HTTP health check test-hc healthCheckNodePort: 12345
W0604 12:56:56.222293 3036007 gce_loadbalancer_external.go:709] Health check test-hc exists but parameters have drifted - updating...
W0604 12:56:56.228103 3036007 gce_loadbalancer_external.go:709] Health check test-hc exists but parameters have drifted - updating...
I0604 12:56:56.234701 3036007 gce_loadbalancer_external.go:155] ensureExternalLoadBalancer(a(/fakesvc)): Ensured IP address 1.2.3.0 (tier: Premium).
I0604 12:56:56.234840 3036007 gce_loadbalancer_external.go:189] ensureExternalLoadBalancer(a(/fakesvc)): Creating firewall.
I0604 12:56:56.234936 3036007 gce_loadbalancer_external.go:193] ensureExternalLoadBalancer(a(/fakesvc)): Created firewall.
I0604 12:56:56.235014 3036007 gce_loadbalancer_external.go:202] ensureExternalLoadBalancer(a(/fakesvc)): Target pool for service doesn't exist.
I0604 12:56:56.235143 3036007 gce_loadbalancer_external.go:901] Creating firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.235228 3036007 gce_loadbalancer_external.go:905] Created firewall k8s-test-cluster-id-node-http-hc for health checks.
I0604 12:56:56.235295 3036007 gce_loadbalancer_external.go:694] Did not find health check k8s-test-cluster-id-node, creating port 10256 path /healthz
I0604 12:56:56.235369 3036007 gce_loadbalancer_external.go:703] Created HTTP health check k8s-test-cluster-id-node healthCheckNodePort: 10256
I0604 12:56:56.235402 3036007 gce_loadbalancer_external.go:553] Creating targetpool a with 1 healthchecks
I0604 12:56:56.235475 3036007 gce_loadbalancer_external.go:495] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created health checks k8s-test-cluster-id-node.
I0604 12:56:56.235502 3036007 gce_loadbalancer_external.go:498] ensureTargetPoolAndHealthCheck(a(/fakesvc)): Created target pool.
I0604 12:56:56.235524 3036007 gce_loadbalancer_external.go:262] ensureExternalLoadBalancer(a(/fakesvc)): Creating forwarding rule, IP 1.2.3.0 (tier: Premium).
I0604 12:56:56.235717 3036007 gce_loadbalancer_external.go:271] ensureExternalLoadBalancer(a(/fakesvc)): Created forwarding rule, IP 1.2.3.0.
I0604 12:56:56.235780 3036007 gce_loadbalancer_external.go:131] ensureExternalLoadBalancer(a(/fakesvc)): Released static IP 1.2.3.0.
W0604 12:56:56.311329 3036007 gce_loadbalancer_internal.go:564] Limiting number of VMs for InstanceGroup k8s-ig--test-cluster-id to 1000
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/gce	0.729s
```
</details>

also the logs from this failure are pretty low signal/noise ...


xref: https://github.com/kubernetes/kubernetes/issues/102607#issuecomment-854985308

--- a/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_test.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_test.go
@@ -185,8 +185,10 @@ func TestBasePath(t *testing.T) {
 	gce, err := fakeGCECloud(vals)
 	// Loadbalancer controller code expects basepath to contain the projects string.
 	expectBasePath := "https://compute.googleapis.com/compute/v1/projects/"
+	// See https://github.com/kubernetes/kubernetes/issues/102757, the endpoint can have mtls in some cases.
+	expectMtlsBasePath := "https://compute.mtls.googleapis.com/compute/v1/projects/"
 	require.NoError(t, err)
-	if gce.service.BasePath != expectBasePath {
-		t.Errorf("Compute basePath has changed. Got %q, want %q", gce.service.BasePath, expectBasePath)
+	if gce.service.BasePath != expectBasePath && gce.service.BasePath != expectMtlsBasePath {
+		t.Errorf("Compute basePath has changed. Got %q, want %q or %q", gce.service.BasePath, expectBasePath, expectMtlsBasePath)
 	}
 }

